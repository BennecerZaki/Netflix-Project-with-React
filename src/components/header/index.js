import React, { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import { HOME } from "../../constants/routes";
import {
  Background,
  ButtonLink,
  Container,
  Dropdown,
  Feature,
  FeatureCallOut,
  Group,
  Link,
  Logo,
  Picture,
  PlayButton,
  Profile,
  Search,
  SearchIcon,
  SearchInput,
  Text,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive(!searchActive)}>
        <img src="/images/icons/search.png" alt="Search Icon" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
      />
    </Search>
  );
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ children, ...restProps }) {
  return (
    <ReactRouterLink to={HOME}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};
