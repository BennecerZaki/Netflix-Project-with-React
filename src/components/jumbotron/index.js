import React from "react";
import { Inner } from "./styles/jumbotron";

const Jumbotron = () => (
  <Inner>
    <p>Hello Again</p>
  </Inner>
);

export default Jumbotron;
