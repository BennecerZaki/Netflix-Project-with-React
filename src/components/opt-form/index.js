import React from "react";
import { Break, Button, Container, Input, Text } from "./styles/opt-form";

function OptForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

OptForm.Input = function OptFormInput({ children, ...restProps }) {
  return <Input {...restProps} />;
};

OptForm.Button = function OptFormButton({ children, ...restProps }) {
  return (
    <Button {...restProps}>
      {children} <img src="/images/icons/chevron-right.png" alt="TryNow!" />
    </Button>
  );
};

OptForm.Text = function OptFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function OptFormBreak({ children, ...restProps }) {
  return <Break {...restProps} />;
};

export default OptForm;
