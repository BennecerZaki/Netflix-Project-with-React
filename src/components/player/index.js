import React, { createContext, useContext, useState } from "react";
import ReactDOM from "react-dom";
import { Button, Close, Container, Inner, Overlay } from "./styles/player";

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);
  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} data-testid="player">
          <Inner>
            <video id="netflix-player" controls>
              <source src={src} type="video/mp4" />
            </video>
            <Close />
          </Inner>
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);
  return (
    <Button onClick={() => setShowPlayer(!showPlayer)} {...restProps}>
      Play
    </Button>
  );
};
