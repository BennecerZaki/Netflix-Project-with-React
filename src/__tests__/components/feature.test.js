import React from "react";
import { render } from "@testing-library/react";
import { Feature } from "../../components";

describe("<Feature/>", () => {
  it("rendres the <Feature /> with populated Data", () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
        <Feature.SubTitle>Watch anywhere. Cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(getByText("Unlimited films, TV programmes and more.")).toBeTruthy();
    expect(getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("rendres the <Feature /> with only title", () => {
    const { container, getByText, queryByText } = render(
      <Feature>
        <Feature.Title>Unlimited films, TV programmes and more.</Feature.Title>
      </Feature>
    );

    expect(getByText("Unlimited films, TV programmes and more.")).toBeTruthy();
    expect(queryByText("Watch anywhere. Cancel at any time.")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("rendres the <Feature /> with only subtitle", () => {
    const { container, getByText, queryByText } = render(
      <Feature>
        <Feature.SubTitle>Watch anywhere. Cancel at any time.</Feature.SubTitle>
      </Feature>
    );

    expect(queryByText("Unlimited films, TV programmes and more.")).toBeFalsy();
    expect(getByText("Watch anywhere. Cancel at any time.")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
