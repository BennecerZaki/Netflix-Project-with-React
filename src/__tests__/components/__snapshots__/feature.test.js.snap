// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature/> rendres the <Feature /> with a bunny video 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 dUyrwq"
>
  <h2
    class="feature__Title-sc-6swrp6-1 jiOEhx"
  >
    Unlimited films, TV programmes and more.
  </h2>
  <h3
    class="feature__SubTitle-sc-6swrp6-2 dQnJQV"
  >
    Watch anywhere. Cancel at any time.
  </h3>
</div>
`;

exports[`<Feature/> rendres the <Feature /> with only subtitle 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 dUyrwq"
>
  <h3
    class="feature__SubTitle-sc-6swrp6-2 dQnJQV"
  >
    Watch anywhere. Cancel at any time.
  </h3>
</div>
`;

exports[`<Feature/> rendres the <Feature /> with only title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 dUyrwq"
>
  <h2
    class="feature__Title-sc-6swrp6-1 jiOEhx"
  >
    Unlimited films, TV programmes and more.
  </h2>
</div>
`;

exports[`<Feature/> rendres the <Feature /> with populated Data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 dUyrwq"
>
  <h2
    class="feature__Title-sc-6swrp6-1 jiOEhx"
  >
    Unlimited films, TV programmes and more.
  </h2>
  <h3
    class="feature__SubTitle-sc-6swrp6-2 dQnJQV"
  >
    Watch anywhere. Cancel at any time.
  </h3>
</div>
`;
